### ISSUE TITLE
pressuredbased activation model for TNT

### ISSUE THREAD

**Issue by cunconbkhp** [synthetik-technologies/blastfoam#4]
Hello! 

I'm trying to run some blastfoam turorials. 

In the 3D buldind case, in setFieldsdict, we have to init the velocity above the explosive, could  you please clarify a bit more about the purpose of this setting . Is this for press the explosive mass to trigger the explosion by pressure and how do you chose the velocity of -400 m/s. 

What will happen if the volume of the the backup cylinder is larger than the volume of the cylinder in regiontocell  setting. By example, i want to set 5 kg of C4  but if  my cylinder is not big enough to cover some grid cells, the backup cylinder will be used. However in this case, the mass of c4 will be larger than expected. In your tuoril : if the first cylinder is used : about 25 kg of c4 but if the backup cylinder is used : >200 kg.

I'm trying to rerun the same case with tnt but i don't know how to calcul the coefficicent (G1,G2,a,x ... ) for TNT in pressurebased mode. Could you please give me a hint?

Thanks in advance!

**Comment by jheylmun** [comment]
There is no specific rational for setting the velocity to -400 m/s for the tutorial case. The only reason it is that high is just to ensure that it does activate the material. The rate stick validation cases use a similar setup, and that velocity is matched to the paper referenced in the README.

The backup region is only used to refine the mesh when the actual region that is to be set will not be caught by even a single cell. It is not used to set any fields though. On the last pass with setRefinedFields, the original volume is forced to be used, even if there still no cells inside the region. Its only purpose is to find extra cells around the actual volume to refine.

As to the coefficients, they are quite difficult to find unfortunately, and there does not seem to be very many sets of coefficients available. I do not believe that I have seen any for TNT, but that is not to say they aren't out there. I've included the reference I have on both the models and the available coefficients I have found. Hopefully they help.

Souers, P Clark, Steve Anderson, James Mercer, Estella McGuire, and Peter Vitello. “JWL++: A Simple Reactive Flow Code Package for Detonation,” 2000, 5.

Guilkey, J.E., T.B. Harman, and B. Banerjee. “An Eulerian–Lagrangian Approach for Simulating Explosions of Energetic Devices.” Computers & Structures 85, no. 11–14 (June 2007): 660–74. https://doi.org/10.1016/j.compstruc.2007.01.031.

Duran, Adam V., and Veeraraghavan Sundararaghavan. “Modeling of Shock Wave Propagation through Energetic Solid State Composites Using a Taylor-Galerkin Scheme.” In 57th AIAA/ASCE/AHS/ASC Structures, Structural Dynamics, and Materials Conference. American Institute of Aeronautics and Astronautics. Accessed May 26, 2020. https://doi.org/10.2514/6.2016-1172.

Kim, Bohoon, Jungsu Park, Kyung-Cheol Lee, and Jack J. Yoh. “A Reactive Flow Model for Heavily Aluminized Cyclotrimethylene-Trinitramine.” Journal of Applied Physics 116, no. 2 (July 14, 2014): 023512. https://doi.org/10.1063/1.4887811.
